pandas/core/indexes/interval.py

```diff
@@ -712,7 +713,7 @@ def _get_indexer(
             # left/right get_indexer, compare elementwise, equality -> match
             indexer = self._get_indexer_unique_sides(target)
 
-        elif not is_object_dtype(target.dtype):
+        elif not (is_object_dtype(target.dtype) or is_string_dtype(target.dtype)):
```

Comment:
why do we exclude string dtype here?  id expect _get_indexer to fastpath to all-minus-ones